src/sentry/services/hybrid_cloud/project/impl.py

```diff
@@ -113,3 +113,28 @@ def create_project_for_organization(
             )
 
             return serialize_project(project)
+
+    def get_or_create_project_for_organization(
+        self,
+        *,
+        organization_id: int,
+        project_name: str,
+        platform: str,
+        user_id: int,
+        add_org_default_team: Optional[bool] = False,
+    ) -> RpcProject:
+        project_query = Project.objects.filter(
+            organization_id=organization_id,
+            name=project_name,
+            platform=platform,
```

Comment:
You should have a `status=ObjectStatus.ACTIVE` condition here so that you don't get a deleted project.